src/olympia/blocklist/mlbf.py

```diff
@@ -127,34 +142,38 @@ def not_blocked_json(self):
         with storage.open(self._not_blocked_path, 'r') as json_file:
             return json.load(json_file)
 
+    def fetch_not_blocked_json(self):
+        """A safer version of .not_blocked_json that will generate from the db
+        if the cached_property isn't set."""
+        try:
+            return self.not_blocked_json
+        except FileNotFoundError:
+            # when self._not_blocked_path doesn't exist
+            pass
+        # see fetch_blocked_json
+        version_excludes = getattr(
+            self, '_version_excludes', self.fetch_blocked_from_db().keys())
+        self.not_blocked_json = self.hash_filter_inputs(
+            self.fetch_all_addons_from_db(version_excludes))
+        return self.not_blocked_json
+
     @property
-    def stash_path(self):
+    def _stash_path(self):
         return os.path.join(
             settings.MLBF_STORAGE_PATH, self.id, 'stash.json')
 
     @cached_property
     def stash_json(self):
```

Comment:
Nit: Should this maybe be `stashed_json` or `json_stash` instead of `stash_json`? `stash_json` sounds like a method, not a property.

Reply:
it's the same naming scheme as `_blocked_path`, `blocked_json` (which reads `blocked.json`).  Could though?